internal/alerts/example-alert-template.json

```diff
@@ -251,6 +251,24 @@
                                 "actionGroupId": "/subscriptions/13d371f9-5a39-46d5-8e1b-60158c49db84/resourceGroups/ContainerInsightsPrometheusCollector-Prod/providers/microsoft.insights/actiongroups/AMA-PrometheusCollectorAlertGroup"
                             }
                         ]
+                    },
+                    {
+                        "alert": "New agent version found for prometheus collector",
+                        "expression": "count(count (kube_pod_container_info{image=~\"mcr.microsoft.com/azuremonitor/containerinsights/ciprod/prometheus-collector.*\"})  by (image)) > 2",
```

Comment:
hi soham - why count of count? just count would do, isn't it ?

Reply:
Thanks Vishwa! I have attached the screenshot of alert query with single count. Since we need only the distinct count of prom collector agent version, we have to use another count on top of the above query. The single count query shows the number of pods with each agent version, but we don't need that count.